File: src/ui/Label.lua
-- Skelly UI element: text label
--
-- By Chris Herborth (https://github.com/Taffer)
-- MIT license, see LICENSE.md for details.

local Class = require 'lib/middleclass/middleclass'
local UIBase = require 'src/ui/UIBase'

-- Label class
local Label = Class('Label', UIBase)

function Label:initialize(parent, x, y, text, font, color, align)
    UIBase.initialize(self, parent)

    self.orig_x = x
    self.x = x
    self.y = y
    self.text = text
    self.font = font
    self.color = color
    self.align = align

    self.width = font:getWidth(self.text)
    self.height = font:getHeight() * font:getLineHeight()

    -- Default is left-align, doesn't require any additional work.
    if align == 'right' then
        self.x = self.x - self.width
    elseif align == 'centre' then
        self.x = self.x - self.width / 2
    end
end

function Label:draw()
    love.graphics.setColor(unpack(self.color))
    love.graphics.setFont(self.font)
    love.graphics.print(self.text, self.x, self.y)
end

function Label:setColor(color)
    self.color = color
end

function Label:setText(text)
    self.text = text
    self.width = self.font:getWidth(self.text)

    if align == 'right' then
        self.x = self.orig_x - self.width
    elseif align == 'centre' then
        self.x = self.orig_x - self.width / 2
    end
end

return Label

File: src/ui/Label.py
# Skelly UI element: text label
#
# By Chris Herborth (https://github.com/Taffer)
# MIT license, see LICENSE.md for details.

import pygame
import pygame.freetype

from .UIBase import UIBase


class Label(UIBase):
    def __init__(self, x: int, y: int, text: str, font: pygame.freetype.Font, color: pygame.Color, align: str) -> None:
        super().__init__()

        self.orig_x = x
        self.x = x
        self.y = y
        self.font = font
        self.color = color
        self.align = align

        self.texture = None
        self.rect = None

        self.set_text(text)

    def draw(self) -> None:
        pygame.display.get_surface().blit(self.texture, self.rect)

    def set_text(self, text: str) -> None:
        self.texture, self.rect = self.font.render(text, self.color)

        if self.align == 'right':
            self.x = self.orig_x - self.rect.width
        elif self.align == 'centre' or self.align == 'center':
            self.x = self.orig_x - self.rect.width / 2

        self.rect.left = self.x
        self.rect.top = self.y
